README.md

```diff
@@ -11,6 +11,103 @@ go get github.com/moolitayer/golang/glog # Dependency
 go get github.com/moolitayer/awx-client-go/awx
 ```
 
+## Usage
+### import
+```
+import 	"github.com/moolitayer/awx-client-go/awx"
+```
+
+### Creating a connection:
+```
+# Uses the builder pattern:
+connection, err := awx.NewConnectionBuilder().
+  Url("http://awx.example.com/api").
+  Username(username).
+  Password(password).
+  Token("TOKEN").
+  Bearer("BEARER").
+  CAFile("/etc/pki/tls/cert.pem").
+  Insecure(insecure).
+  Proxy(http://myproxy.example.com).
+  Build()                                      # Create the client
+if err != nil {
+  panic(err)
+}
+defer connection.Close()                      # Don't forget to close the connection!
+```
+`Url()` points at an AWX server's root API endpoint.
+`Proxy()` specifies A proxy server to use for all outgoing connection to the AWX server.
+#### Authentication
+`Username()` and `Password()` specify Basic Auth for AWX API server.
+`Token()` uses the authtoken/ endpoint and works with AWX XXX and Ansible tower < 3.3.
+`Bearer()` uses OAuth2 and works since AWX XXX and Ansible Tower > 3.3.
+
+#### TLS
+`CAFile()` specifies a trusted PEM encoded CA certificates used to verify the AWX server.
```

Comment:
Nit: this specifies the path to the file, not the PEM-encoded text. Obvious from the name, but the description doesn't much. Also maybe mention that the host trust store will be used if no CA is provided.